### PR TITLE
SDL 0119 - SDL Passenger Mode

### PR DESCRIPTION
**Pull request by Tnnnguyen**
Fixes # [SDL-0119](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0119-SDL-passenger-mode.md)

Revised to accommodate changes from [Revise SDL-0119 SDL Passenger Mode](https://github.com/smartdevicelink/sdl_evolution/pull/775)

This PR is **[ready]** for review.

### Risk
This PR makes **[no]** API changes.

### Testing Plan
Test with the new Boolean bit from OnDriverDistraction, if it's true, the lockscreen can be dismissed.

### Summary
The lockscreen can be swiped to dismiss if dismissible

### CLA
- [x ] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by Tnnnguyen** [comment]
@theresalech What is the exact wording of the additional message that you want? Also where is the new message's location within the lock screen?

**Comment by Tnnnguyen** [comment]
@theresalech Do you have the exact wording of the additional message that you want? Also where is the new message's location within the lock screen?

**Comment by theresalech** [comment]
@Tnnnguyen there is a proposal in review to confirm this wording: https://github.com/smartdevicelink/sdl_evolution/issues/769.

**Comment by Tnnnguyen** [comment]
@theresalech thanks.

**Comment by theresalech** [comment]
@Tnnnguyen the lock screen dismissal message can be at the bottom of the screen, replacing the "Locked for your safety" text.  You can see the iOS implementation for reference: https://github.com/smartdevicelink/sdl_ios/pull/1296.

**Comment by Tnnnguyen** [comment]
@theresalech thanks, I kind of guess, so I made that change on my last revision.

**Comment by Tnnnguyen** [comment]
@theresalech this PR is ready for review, thanks.

File: android/sdl_android/src/main/java/com/smartdevicelink/managers/lockscreen/LockScreenManager.java [hunk 1]
```diff
@@ -77,6 +79,10 @@
 	protected boolean lockScreenEnabled, deviceLogoEnabled;
 	protected int lockScreenIcon, lockScreenColor, customView;
 	protected Bitmap deviceLogo;
+	private boolean mIsLockscreenDissmissible;
```
Comment (BrettyWhite) [inline android/sdl_android/src/main/java/com/smartdevicelink/managers/lockscreen/LockScreenManager.java]: spelling

File: android/sdl_android/src/main/java/com/smartdevicelink/managers/lockscreen/LockScreenManager.java [hunk 2]
```diff
@@ -166,6 +177,14 @@ public void onNotified(RPCNotification notification) {
 				// do something with the status
 				if (notification != null) {
 					OnDriverDistraction ddState = (OnDriverDistraction) notification;
+					Boolean isDismissable = ddState.getLockscreenDismissibility();
```
Comment (BrettyWhite) [inline android/sdl_android/src/main/java/com/smartdevicelink/managers/lockscreen/LockScreenManager.java]: spelling

File: android/sdl_android/src/main/java/com/smartdevicelink/managers/lockscreen/LockScreenManager.java [hunk 3]
```diff
@@ -166,6 +177,14 @@ public void onNotified(RPCNotification notification) {
...
+					if (isDismissable != null) {
+						mIsLockscreenDissmissible = isDismissable;
```
Comment (BrettyWhite) [inline android/sdl_android/src/main/java/com/smartdevicelink/managers/lockscreen/LockScreenManager.java]: spelling

File: base/src/main/java/com/smartdevicelink/proxy/rpc/OnDriverDistraction.java [hunk 4]
```diff
@@ -73,6 +73,8 @@
  */
 public class OnDriverDistraction  extends RPCNotification {
 	public static final String KEY_STATE = "state";
+    public static final String KEY_LOCKSCREEN_DISMISSIBLE = "lockScreenDismissalEnabled";
```
Comment (BrettyWhite) [inline base/src/main/java/com/smartdevicelink/proxy/rpc/OnDriverDistraction.java]: can you fix the indentations to match?

File: base/src/main/java/com/smartdevicelink/proxy/rpc/OnDriverDistraction.java [hunk 5]
```diff
@@ -107,5 +109,37 @@ public DriverDistractionState getState() {
      */    
     public void setState( @NonNull DriverDistractionState state ) {
         setParameters(KEY_STATE, state);
-    }  
+    }
+
+    /**
+     * <p>Called to set dismissible state of Lockscreen</p>
+     * @param isDismissible the Lockscreen'S dismissibility
```
Comment (BrettyWhite) [inline base/src/main/java/com/smartdevicelink/proxy/rpc/OnDriverDistraction.java]:
```suggestion
     * @param isDismissible the Lockscreen's dismissibility
```

File: base/src/main/java/com/smartdevicelink/proxy/rpc/OnDriverDistraction.java [hunk 6]
```diff
@@ -107,5 +109,37 @@ public DriverDistractionState getState() {
...
+     */
+    public void setLockscreenDismissible(boolean isDismissible) {
+        setParameters(KEY_LOCKSCREEN_DISMISSIBLE, isDismissible);
+    }
+
+    /**
+     * <p>Called to get the dismissible state of Lockscreen</p>
+     * @return true if the Lockscreen is dismissible, false otherwise
+     */
+    public Boolean getLockscreenDismissibility() {
```
Comment (BrettyWhite) [inline base/src/main/java/com/smartdevicelink/proxy/rpc/OnDriverDistraction.java]: The setter and getter method naming should match for the same key

File: android/sdl_android/src/main/java/com/smartdevicelink/managers/lockscreen/SDLLockScreenActivity.java [hunk 7]
```diff
@@ -94,6 +104,15 @@ protected void onCreate(Bundle savedInstanceState) {
...
 	@Override
+	public boolean onTouchEvent(MotionEvent event) {
+		if (mIsDismissible) {
+			return mGestureDetector.onTouchEvent(event);
+		}
+		return super.onTouchEvent(event);
+	}
+
+
+		@Override
```
Comment (BrettyWhite) [inline android/sdl_android/src/main/java/com/smartdevicelink/managers/lockscreen/SDLLockScreenActivity.java]: double space and indentation

File: android/sdl_android/src/main/java/com/smartdevicelink/managers/lockscreen/LockScreenManager.java (hunk 1 above)
Comment (Tnnnguyen) [reply android/sdl_android/src/main/java/com/smartdevicelink/managers/lockscreen/LockScreenManager.java]: Done

File: android/sdl_android/src/main/java/com/smartdevicelink/managers/lockscreen/LockScreenManager.java (hunk 2 above)
Comment (Tnnnguyen) [reply android/sdl_android/src/main/java/com/smartdevicelink/managers/lockscreen/LockScreenManager.java]: Done

File: base/src/main/java/com/smartdevicelink/proxy/rpc/OnDriverDistraction.java (hunk 4 above)
Comment (Tnnnguyen) [reply base/src/main/java/com/smartdevicelink/proxy/rpc/OnDriverDistraction.java]: Done, no change, mis-alignment is from previous code.

File: base/src/main/java/com/smartdevicelink/proxy/rpc/OnDriverDistraction.java (hunk 5 above)
Comment (Tnnnguyen) [reply base/src/main/java/com/smartdevicelink/proxy/rpc/OnDriverDistraction.java]: Done

File: base/src/main/java/com/smartdevicelink/proxy/rpc/OnDriverDistraction.java (hunk 6 above)
Comment (Tnnnguyen) [reply base/src/main/java/com/smartdevicelink/proxy/rpc/OnDriverDistraction.java]: Done

File: android/sdl_android/src/main/java/com/smartdevicelink/managers/lockscreen/SDLLockScreenActivity.java (hunk 7 above)
Comment (Tnnnguyen) [reply android/sdl_android/src/main/java/com/smartdevicelink/managers/lockscreen/SDLLockScreenActivity.java]: Done

**Pull request merged**